packages/p2p/src/components/modal-manager/modal-manager.tsx

```diff
@@ -1,43 +1,51 @@
 import React from 'react';
-import { modals } from 'Constants/modals';
+import { Modals, TModalKeys, TModalProps, TModals } from 'Constants/modals';
 import { useModalManagerContext } from './modal-manager-context';
+import { TModal } from './modal-manager-context-provider';
+
+type LazyModal = React.LazyExoticComponent<React.ComponentType<TModalProps[TModalKeys]>>;
 
 const ModalManager = () => {
     const { modal, modal_props, stacked_modal } = useModalManagerContext();
+    // type guard for nullish modal value, exit early
+    if (!modal) return null;
 
     const { key } = modal;
-    const Modal = modals[key];
-    const StackedModal = modals[stacked_modal?.key];
+    const Modal = Modals[key] as LazyModal;
+
+    const StackedModal = stacked_modal ? (Modals[stacked_modal.key] as LazyModal) : null;
 
-    const getModalProps = current_modal => {
+    const getModalProps = <T extends TModalKeys>(current_modal: TModal<T>): TModalProps[T] => {
         if (current_modal?.props && Object.keys(current_modal.props).length > 0) {
             // if props was provided to the argument and it was also already initialised using useRegisterModalProps,
             // merge the 2 props together and update latest prop values with the passed prop argument
             if (modal_props.has(current_modal.key)) {
                 return {
                     ...modal_props.get(current_modal.key),
                     ...current_modal.props,
-                };
+                } as TModalProps[T];
             }
             return current_modal.props;
         }
         if (modal_props.has(current_modal.key)) {
-            return modal_props.get(current_modal.key);
+            return modal_props.get(current_modal.key) as TModalProps[T];
```

Comment:
Map `get()` method returns the generic argument `V` from `Map<K, V>`, and `modal_props` is typed as `Map<TModalKeys, TModalProps[TModalKeys]>`, thus `get()` would return `TModalProps[TModalKeys` and not as `TModalProps[T]`, thus I had to cast it as `TModalProps[T]`
![Screenshot 2023-07-05 at 2 13 11 PM](https://github.com/farrah-deriv/deriv-app/assets/103016120/61c6fe5c-87d6-48fe-b5d2-9a683dc13b8d)